File: Sources/XMTPiOS/Messages/DecryptedMessage.swift
//
//  DecryptedMessage.swift
//
//
//  Created by Pat Nakajima on 11/14/23.
//

import Foundation

public struct DecryptedMessage {
	public var id: String
	public var encodedContent: EncodedContent
	public var senderAddress: String
	public var sentAt: Date
	public var topic: String = ""
}

File: library/src/main/java/org/xmtp/android/library/messages/DecryptedMessage.kt
package org.xmtp.android.library.messages

import org.xmtp.android.library.codecs.EncodedContent
import java.util.Date

data class DecryptedMessage(
    var id: String,
    var encodedContent: EncodedContent,
    var senderAddress: String,
    var sentAt: Date,
    var topic: String = "",
    var deliveryStatus: MessageDeliveryStatus = MessageDeliveryStatus.PUBLISHED
)
